src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -57,30 +60,52 @@ public ProductReturnResponseDto getProductReturnResponseDto(
 
   public List<ProductResponseDto> getAllProducts() {
     List<Product> products = productRepository.findAll();
+    logger.debug("Fetching all products");
     return products.stream().map(productMapper::mapToProductResponseDto).toList();
   }
 
   public List<ProductResponseDto> getByOwner(UUID ownerId) {
     List<Product> products = productRepository.findAllByOwnerId(ownerId);
+    logger.info("Get product by owner with ID: {}", ownerId);
     return products.stream().map(productMapper::mapToProductResponseDto).toList();
   }
 
   public Product getProduct(UUID id) {
+    logger.info("Get product by ID: {}", id);
     return productRepository.findById(id).orElseThrow(() -> new ProductNotFoundException(id));
   }
 
   public ProductResponseDto getProductResponse(UUID id) {
+    logger.info("Get productResponse by ID: {}", id);
     return productMapper.mapToProductResponseDto(getProduct(id));
   }
 
   public void updateProductOwnerAndSale(Product product, User newOwner, Sale sale) {
     updateProductOwnerRecursively(product, newOwner);
-    product.setPartOfSale(sale);
+    if (sale != null) {
+      product.setPartOfSale(sale);
+      logger.debug(
+          "Updated product owner and sale for product with ID: {}. New owner with ID: {}, Sale with ID: {}",
+          product.getId(),
+          newOwner.getId(),
+          sale.getId());
+    } else {
+      product.setPartOfSale(null);
+      logger.debug(
+          "Updated product owner without sale for product with ID: {}. New owner with ID: {}, Sale set to null",
+          product.getId(),
+          newOwner.getId());
+    }
```

Comment:
This looks like an overkill. You can do something in the lines of:
```suggestion
      product.setPartOfSale(sale);
      logger.debug(
          "Updated product owner and sale for product with ID: {}. New owner with ID: {}, Sale with ID: {}",
          product.getId(),
          newOwner.getId(),
          sale == null ? null : sale.getId();
```